infra/modules/azure_core_infra/_modules/apim/variables.tf

```diff
@@ -0,0 +1,109 @@
+variable "project" {
+  type        = string
+  description = "env prefix, short environment and short location"
+}
+
+variable "prefix" {
+  type        = string
+  description = "env prefix, short environment and short location amd domain"
+}
+
+variable "suffix" {
+  type        = string
+  description = "the instance number"
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "Resource group name"
+}
+
+variable "location" {
+  type        = string
+  description = "Location"
+}
+
+variable "tags" {
+  type        = map(any)
+  description = "Resources tags"
+}
+
+variable "tier" {
+  type        = string
+  description = "Resource tiers depending on demanding workload. Allowed values are 's', 'm', 'l'."
+  default     = "s"
+
+  validation {
+    condition     = contains(["s", "m", "l"], var.tier)
+    error_message = "Allowed values for \"tier\" are \"s\", \"m\", or \"l\"."
+  }
+}
+
+variable "publisher_name" {
+  type        = string
+  description = "The name of publisher/company."
+}
+
+variable "publisher_email" {
+  type        = string
+  description = "The email of publisher/company."
+}
+
+variable "enable_public_network_access" {
+  type        = bool
+  description = "Enable public network access"
+  default     = false
+}
+
+variable "virtual_network_type_internal" {
+  type        = bool
+  description = "The type of virtual network you want to use, if true it will be Internal and you need to specify a subnet_id, otherwise it will be None"
```

Comment:
```suggestion
  description = "If set to true, the virtual network will be internal, requiring you to specify a subnet_id. Otherwise, it will be None"
```